server/jwt/jwt.go

```diff
@@ -0,0 +1,162 @@
+package jwt
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/fabiocicerchia/go-proxy-cache/config"
+	"github.com/lestrrat-go/backoff/v2"
+	"github.com/lestrrat-go/jwx/jwk"
+	"github.com/lestrrat-go/jwx/jwt"
+	log "github.com/sirupsen/logrus"
+)
+
+var co *config.Jwt
+var jwtKeyFetcher *jwk.AutoRefresh
+
+func InitJWT(conifg *config.Jwt) {
+
+	if co == nil {
+		co = conifg
+		jwtKeyFetcher = jwk.NewAutoRefresh(co.Context)
+		jwtKeyFetcher.Configure(co.Jwks_url,
+			jwk.WithMinRefreshInterval(15*time.Minute),
+			jwk.WithFetchBackoff(backoff.Constant(backoff.WithInterval(time.Minute))),
+		)
+	}
+}
+
+func InitJWTWithDomainConf(domainConfig config.Configuration) {
+	var jwtUrl string
+	var allowedScopes, includedPaths []string
+	if domainConfig.Jwt.Jwks_url != "" {
+		jwtUrl = domainConfig.Jwt.Jwks_url
+	} else {
+		jwtUrl = config.Config.Jwt.Jwks_url
+	}
+	if domainConfig.Jwt.Allowed_scopes != nil {
+		allowedScopes = domainConfig.Jwt.Allowed_scopes
+	} else {
+		allowedScopes = config.Config.Jwt.Allowed_scopes
+	}
+	if domainConfig.Jwt.Included_paths != nil {
+		includedPaths = domainConfig.Jwt.Included_paths
+	} else {
+		includedPaths = config.Config.Jwt.Included_paths
+	}
```

Comment:
suggesting of skipping the `else`:

```suggestion
	jwtUrl = config.Config.Jwt.Jwks_url
	if domainConfig.Jwt.Jwks_url != "" {
		jwtUrl = domainConfig.Jwt.Jwks_url		
	}
	allowedScopes = config.Config.Jwt.Allowed_scopes
	if domainConfig.Jwt.Allowed_scopes != nil {
		allowedScopes = domainConfig.Jwt.Allowed_scopes
	}
	includedPaths = config.Config.Jwt.Included_paths
	if domainConfig.Jwt.Included_paths != nil {
		includedPaths = domainConfig.Jwt.Included_paths
	}
```

Reply:
done!